src/hotspot/share/compiler/compilationPolicy.hpp

```diff
@@ -34,11 +34,21 @@ class CompileTask;
 class CompileQueue;
 /*
  *  The system supports 5 execution levels:
- *  * level 0 - interpreter
+ *  * level 0 - interpreter (Profiling is tracked by a MethodData object, or MDO in short)
  *  * level 1 - C1 with full optimization (no profiling)
  *  * level 2 - C1 with invocation and backedge counters
- *  * level 3 - C1 with full profiling (level 2 + MDO)
- *  * level 4 - C2
+ *  * level 3 - C1 with full profiling (level 2 + All other MDO profiling information)
+ *  * level 4 - C2 with full profile guided optimization
+ *
+ * The MethodData object is used by both the interpreter and C1 to store any profiling
+ * information collected on a method (CompilationPolicy::create_mdo, compilationPolicy.cpp
+ * for the interpreter and ciMethod::ensure_method_data, ciMethod.cpp for C1), and interacts
+ * with C1 and C2 via the compiler interface. It is updated periodically as more profiling
+ * information is gathered, directly in the case of the interpreter and through ciMethodData
```

Comment:
This is still a bit misleading.  The information flow between MethodData and ciMethodData is one-way.  Only MethodData are updated by the interpreter or generated code.  ciMethodData is just a read-only snapshot used during compilation.

Reply:
Revised